brooklyn-server/server-cli/src/main/java/org/apache/brooklyn/cli/ItemLister.java

```diff
@@ -98,88 +115,150 @@ public Void call() throws Exception {
-                    String type = (String) policy.get("type");
-                    String name = (String) policy.get("name");
-                    String policyHtml = TemplateProcessor.processTemplateContents(policyTemplateHtml, ImmutableMap.of("type", type, "name", name));
-                    Files.write(policyHtml, new File(Os.mergePaths(outputFolder, "policies", type + ".html")), Charsets.UTF_8);
-                }
-                
-                List<Map<String, Object>> enrichers = (List<Map<String, Object>>) result.get("enrichers");
-                String enricherTemplateHtml = resourceUtils.getResourceAsString(Urls.mergePaths(BASE_TEMPLATES, "enricher.html"));
-                for (Map<String, Object> enricher : enrichers) {
-                    String type = (String) enricher.get("type");
-                    String name = (String) enricher.get("name");
-                    String enricherHtml = TemplateProcessor.processTemplateContents(enricherTemplateHtml, ImmutableMap.of("type", type, "name", name));
-                    Files.write(enricherHtml, new File(Os.mergePaths(outputFolder, "enrichers", type + ".html")), Charsets.UTF_8);
+                for (URL url: urls) {
+                    List<Map<?,?>> entities = new ArrayList<>();
+                    List<Map<?,?>> policies = new ArrayList<>();
+                    List<Map<?,?>> enrichers = new ArrayList<>();
+                    List<Map<?,?>> locations = new ArrayList<>();
+                    List<Map<?,?>> locationResolvers = new ArrayList<>();
+
+                    String yamlContent = Files.toString(new File(url.getFile()), Charset.forName("UTF-8"));
+
+                    catalog.addItems(yamlContent);
+                    for (CatalogItem item: catalog.getCatalogItems()) {
+                        AbstractBrooklynObjectSpec<?,?> spec = catalog.createSpec(item);
+                        Map<String,Object> itemDescriptor = ItemDescriptors.toItemDescriptor((Class<? extends BrooklynObject>) spec.getType(), false);
+                        List<EntityConfigSummary> config = new ArrayList<>();
+
+                        if (!itemDescriptor.containsKey("name")) {
+                            itemDescriptor.put("name", item.getDisplayName());
+                        }
+                        if (!itemDescriptor.containsKey("description")) {
+                            itemDescriptor.put("description", item.getDescription());
+                        }
+                        if (!itemDescriptor.containsKey("iconUrl")) {
+                            itemDescriptor.put("iconUrl", item.getIconUrl());
+                        }
+                        for (SpecParameter<?> param: spec.getParameters()) {
+                            EntityConfigSummary entityConfigSummary = EntityTransformer.entityConfigSummary(param.getType(),
+                                    param.getLabel(), null, MutableMap.<String,URI>of());
```

Comment:
Apart for ordering `priority` can be used to decide on whether an item should be shown to the user.
Create a counter here, passing its value when `param.isPinned()`, `null` otherwise.